File: Data Structures & Algorithms/find-duplicate-integer/submission-3.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        fp = 0
        sp = 0

        while True:
            fp = nums[fp]
            fp = nums[fp]
            sp = nums[sp]

            if sp == fp:
                fp = 0
                
                while sp != fp:
                    fp = nums[fp]
                    sp = nums[sp]

                    if fp == sp:
                        return fp
        return -1

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.java
class Solution {
    public int longestConsecutive(int[] nums) {

        // Lets make a HashSet first

        HashSet<Integer> numsMap = new HashSet<>();

        for(int num : nums){
            numsMap.add(num);
        }

        int maxStreak = 0;

        for(int num : nums){
            if(!numsMap.contains(num-1)){
                int length = 1;
                int curr = num;

                while(numsMap.contains(curr+1)){
                    curr = curr+1;
                    length+=1;
                }

                maxStreak = Math.max(maxStreak, length);
            }
        }

        return maxStreak;
        
    }
}
